ssl/ssl_key_share.cc

```diff
@@ -386,6 +583,23 @@ bool ssl_name_to_group_id(uint16_t *out_group_id, const char *name, size_t len)
   return false;
 }
 
+bool  get_ec_public_key_length(uint16_t *out_public_key_length, int nid) {
+  if (nid == NID_X25519) {
+    *out_public_key_length = 32;
+    return true;
+  }
+
+  EC_GROUP *ec_group = EC_GROUP_new_by_curve_name(nid);
+  if (!ec_group) {
+    return false;
+  }
+  const size_t field_len = BN_num_bytes(&ec_group->field);
+  *out_public_key_length = 1 + (2 * field_len);
```

Comment:
why `+1`? it'd be nice to add a comment explaining why is the length computed in this way.